packages/dd-trace/src/appsec/iast/analyzers/vulnerability-analyzer.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const Plugin = require('../../../../src/plugins/plugin')
+const { storage } = require('../../../../../datadog-core')
+const { getFirstNonDDPathAndLine } = require('./../path-line')
+const { createVulnerability, addVulnerability } = require('../vulnerability-reporter')
+const { getIastContext } = require('../iast-context')
+const overheadController = require('../overhead-controller')
+
+class Analyzer extends Plugin {
+  constructor (type) {
+    super()
+    this._type = type
+  }
+
+  _isVulnerable (value, context) {
+    return false
+  }
+
+  _report (value, context) {
+    const evidence = this._getEvidence(value)
+    const location = this._getLocation()
+    const spanId = context && context.rootSpan && context.rootSpan.context().toSpanId()
+    const vulnerability = createVulnerability(this._type, evidence, spanId, location)
+    addVulnerability(context, vulnerability)
+  }
+
+  _getEvidence (value) {
+    return { value }
+  }
+
+  _getLocation () {
+    return getFirstNonDDPathAndLine()
+  }
+
+  analyze (value) {
+    const iastContext = getIastContext(storage.getStore())
+    if (this._isVulnerable(value, iastContext) && this._checkOCE(iastContext)) {
```

Comment:
Currently we don't know how are we going to work out of the request, so we can't report nothing if we have iastContext
```suggest
if (iastContext && this._isVulnerable(value, iastContext) && this._checkOCE(iastContext)) {
```